bin/migrate-oats-data/submissions/app_submissions.py

```diff
@@ -0,0 +1,262 @@
+            page_size=batch_size,
+        )
+
+    if len(inc_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_inc(),
+            inc_data_list,
+            page_size=batch_size,
+        )
+
+    if len(other_data_list) > 0:
+        execute_batch(
+            cursor,
+            get_update_query_for_other(),
+            other_data_list,
+            page_size=batch_size,
+        )
+
+    conn.commit()
+
+def prepare_app_sub_data(app_sub_raw_data_list):
+    """
+    This function prepares different lists of data based on the 'alr_change_code' field of each data dict in 'app_sub_raw_data_list'.
+
+    :param app_sub_raw_data_list: A list of raw data dictionaries.
+    :return: Five lists, each containing dictionaries from 'app_sub_raw_data_list' grouped based on the 'alr_change_code' field
+
+    Detailed Workflow:
+    - Initializes empty lists
+    - Iterates over 'app_sub_raw_data_list'
+        - Maps the basic fields of the data dictionary based on the alr_change_code
+    - Returns the mapped lists
+    """
+    nfu_data_list = []
+    nar_data_list = []
+    exc_data_list = []
+    inc_data_list = []
+    other_data_list = []
+
+    for row in app_sub_raw_data_list:
+        data = dict(row)
+        # data = map_basic_field(data)
+
+        if data["alr_change_code"] == ALRChangeCode.NFU.value:
+            # data = mapOatsToAlcsAppPrep(data)
+            nfu_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.NAR.value:
+            nar_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.EXC.value:
+            # data = mapOatsToAlcsLegislationCode(data)
+            exc_data_list.append(data)
+        elif data["alr_change_code"] == ALRChangeCode.INC.value:
+            # data = mapOatsToAlcsLegislationCode(data)
+            inc_data_list.append(data)
+        else:
+            other_data_list.append(data)
+
+    return nfu_data_list, nar_data_list, other_data_list, exc_data_list, inc_data_list
+
+
+def get_update_query(unique_fields,unique_values):
+    # unique_fields takes input from called function and appends to query
+    query = """
+                INSERT INTO alcs.application_submission (
+                    file_number,
+                    local_government_uuid,
+                    type_code,
+                    is_draft,
+                    audit_created_by,
+                    applicant
+                    {unique_fields}
+                )
+                VALUES (
+                    %(file_number)s,
+                    %(local_government_uuid)s,
+                    %(type_code)s,
+                    false,
+                    'oats_etl',
+                    %(applicant)s
+                    {unique_values}
+                )
+    """
+    return query.format(unique_fields=unique_fields, unique_values=unique_values)
+
+def get_update_query_for_nfu():
+    unique_fields = """, 
```

Comment:
You do not need a pre-formatted string variable. You could use unique_fields = ", nfu_hectares" instead. the same goes for unique_values.

Reply:
when there are multiple values later on it is easier to keep them pre-formatted

Reply:
Agree regarding multiple values. However, since, at this point, you are not adding anything, it is better to keep this simple and add features as we go. That way, it is easier to keep code clean and consistent.